RealmSwift/Tests/RealmTests.swift

```diff
@@ -0,0 +1,462 @@
+
+        // get object
+        let results = Realm().objects(SwiftStringObject.self)
+        XCTAssertEqual(results.count, Int(1), "There should be 1 object of type StringObject")
+        XCTAssertEqual(results[0].stringCol, "string", "Value of first column should be 'string'")
+    }
+
+    func testRefresh() {
+        let realm = Realm()
+        realm.autorefresh = false
+
+        // test that autoreresh is  notapplied
+        // we have two notifications, one for opening the realm, and a second when performing our transaction
+        let notificationFired = expectationWithDescription("notification fired")
+        let token = realm.addNotificationBlock { _, realm in
+            XCTAssertNotNil(realm, "Realm should not be nil")
+            notificationFired.fulfill()
+        }
+
+        let results = realm.objects(SwiftStringObject.self)
+        XCTAssertEqual(results.count, Int(0), "There should be 1 object of type StringObject")
+
+        dispatch_async(dispatch_queue_create("background", nil)) {
+            Realm().write {
+                SwiftStringObject.createInRealm(Realm(), withObject: ["string"])
+                return
+            }
+        }
+        waitForExpectationsWithTimeout(2, handler: nil)
+        realm.removeNotification(token)
+
+        XCTAssertEqual(results.count, Int(0), "There should be 1 object of type StringObject")
+
+        // refresh 
+        realm.refresh()
+
+        XCTAssertEqual(results.count, Int(1), "There should be 1 object of type StringObject")
+        XCTAssertEqual(results[0].stringCol, "string", "Value of first column should be 'string'")
+    }
+
+    func testInvalidate() {
+        let realm = Realm()
+        let object = SwiftObject()
+        realm.write({
+            realm.add(object)
+            return
+        })
+        realm.invalidate()
+        XCTAssertEqual(object.invalidated, true)
+
+        realm.write({
+            realm.add(SwiftObject())
+            return
+        })
+        XCTAssertEqual(realm.objects(SwiftObject).count, 2)
+        XCTAssertEqual(object.invalidated, true)
+    }
+
+    func testWriteCopyToPath() {
+        let realm = Realm()
+        realm.write {
+            realm.add(SwiftObject())
+        }
+        let path = Realm.defaultPath.stringByDeletingLastPathComponent.stringByAppendingPathComponent("copy.realm")
+        XCTAssertNil(realm.writeCopyToPath(path))
+        autoreleasepool {
+            let copy = Realm(path: path)
+            XCTAssertEqual(1, copy.objects(SwiftObject.self).count)
+        }
+        NSFileManager.defaultManager().removeItemAtPath(path, error: nil)
+    }
+
+    func testSetEncryptionKey() {
+        Realm.setEncryptionKey(NSMutableData(length: 64))
+        Realm.setEncryptionKey(nil, forPath: Realm.defaultPath)
+        XCTAssert(true, "setting those keys should not throw")
+    }
+
+    func testEquals() {
+        let realm = Realm()
+        XCTAssertTrue(realm == Realm())
```

Comment:
Does `XCTAssertEqual` not work here, or are you just being explicit about it testing `==`?